cmd/hiffy/src/lib.rs

```diff
@@ -166,23 +164,27 @@ pub fn hiffy_list(hubris: &HubrisArchive, verbose: bool) -> Result<()> {
         let module = hubris.lookup_module(HubrisTask::Task(i as u32))?;
 
         if let Some(iface) = &module.iface {
-            let mut ops = iface.ops.iter();
+            if !filter.is_empty()
+                && !filter.iter().any(|f| iface.name == *f || module.name == *f)
+            {
+                continue;
+            }
 
-            print!("{:15} {:<12} ", module.name, iface.name);
+            let mut ops = iface.ops.iter().peekable();
 
-            match ops.next() {
-                None => {
-                    println!("-");
-                }
-                Some(op) => {
-                    print!("{:<20}", op.0);
-                    print_args(&op, module, 49);
+            println!("{:<28} TASK", "INTERFACE");
+            println!("{:<28} {}", iface.name, module.name);
+            println!("  |");
 
-                    for op in ops {
-                        print!("{:29}{:<20}", "", op.0);
-                        print_args(&op, module, 49);
-                    }
-                }
+            while let Some(op) = ops.next() {
+                println!("  +--> {}.{}", iface.name, op.0);
+
+                let last = ops.peek().is_none();
+                let c = if last { "" } else { "|" };
+                let margin = format!("  {:<8}", c);
+
+                print_args(&op, module, margin);
+                println!("  {}", c);
             }
         }
     }
```

Comment:
It would be nice to print a message if nothing matches, e.g.
```rust
    if !any_matches {
        println!(
            "warning: no matching tasks or interfaces found.  \
             Options are shown below:"
        );
        println!("{:<28} TASK", "INTERFACE");
        for i in 0..hubris.ntasks() {
            let module = hubris.lookup_module(HubrisTask::Task(i as u32))?;
            if let Some(iface) = &module.iface {
                println!("{:<28} {}", iface.name, module.name);
            }
        }
    }
```
(assuming that you have a `any_matches` variable updated in the loop)

Reply:
Yeah, good call, though I would rather explicitly bail in this case.